ONSdigital/ssdc-rm-response-operations
# Make page heading text a link back to homepage

# Motivation and Context
Without any user research, user experience expertise, user interaction design or general UI expert input, doing anything to navigation is somewhat misguided and unwise, but there was a desire for making the text at the top of the page into a link back to the homepage... sounds easy, right?

# What has changed
Totally restructured app so that the homepage link works... this is a ReactJS single-page application (SPA) so "just make that text a link" is easier said than done. We don't want the client browser having to reload all the JS and re-initialise itself every time the user clicks on something, so we _have_ to use the proper routing system, which meant moving almost all the remaining HTML into the ReactJS code.

# How to test?
Click around and see what you think.

There are an ever-growing number of UI horrible problems now, which are hard to sort out without expertise in CSS and the ONS design system, which is very large and complex.

Pre-existing problems, before this PR:

- Header ONS logo and text was out of alignment with the body
- Footer logo and text out of alignment with the body
- Footer was disappearing off the bottom of the screen, even though there was plenty of space for it to fit

Now, there's a _new_ problem where the footer is not 'sticky' at the bottom of the screen... it ends where the content ends, which looks ugly on screens where there's not much content. Technically, this is an improvement on the footer which was never seen before because it was always off the screen, but it _is_ quite ugly and I personally wouldn't be happy to put this into production as-is, even though it doesn't affect usability or accessibility. However, I have resolved the other two problems as part of this PR.

# Links
Trello: https://trello.com/c/7fDN3clB

# Screenshots (if appropriate):
<img width="1144" alt="Screenshot 2021-12-20 at 13 45 13" src="https://user-images.githubusercontent.com/41681172/146777596-5c353bc3-a993-42a7-bc20-747388e5997d.png">
